src/main/java/de/unijena/cheminf/clustering/desccalc/Descriptor.java

```diff
@@ -25,120 +25,1210 @@
+            descriptorToCdkObjectMap.put(VABC, new VABCDescriptor());
+
+            // PUBCHEM_FINGERPRINTER
+            fingerprintToCdkObjectMap.put(PUBCHEM_FINGERPRINTER, new PubchemFingerprinter(SilentChemObjectBuilder.getInstance()));
+
+            // CIRCULAR_FINGERPRINTER_ECFP_0
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_ECFP_0, new CircularFingerprinter(CircularFingerprinter.CLASS_ECFP0, 1024));
+
+            // CIRCULAR_FINGERPRINTER_FCFP_0
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_FCFP_0, new CircularFingerprinter(CircularFingerprinter.CLASS_FCFP0, 1024));
+
+            // CIRCULAR_FINGERPRINTER_ECFP_2
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_ECFP_2, new CircularFingerprinter(CircularFingerprinter.CLASS_ECFP2, 1024));
+
+            // CIRCULAR_FINGERPRINTER_FCFP_2
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_FCFP_2, new CircularFingerprinter(CircularFingerprinter.CLASS_FCFP2, 1024));
+
+            // CIRCULAR_FINGERPRINTER_ECFP_4
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_ECFP_4, new CircularFingerprinter(CircularFingerprinter.CLASS_ECFP4, 1024));
+
+            // CIRCULAR_FINGERPRINTER_FCFP_4
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_FCFP_4, new CircularFingerprinter(CircularFingerprinter.CLASS_FCFP4, 1024));
+
+            // CIRCULAR_FINGERPRINTER_ECFP_6
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_ECFP_6, new CircularFingerprinter(CircularFingerprinter.CLASS_ECFP6, 1024));
+
+            // CIRCULAR_FINGERPRINTER_FCFP_6
+            fingerprintToCdkObjectMap.put(CIRCULAR_FINGERPRINTER_FCFP_6, new CircularFingerprinter(CircularFingerprinter.CLASS_FCFP6, 1024));
+
+            // MACCS_FINGERPRINTER
+            fingerprintToCdkObjectMap.put(MACCS_FINGERPRINTER, new MACCSFingerprinter(SilentChemObjectBuilder.getInstance()));
+
+            // Add new descriptor information here!
+
+        } catch (Exception anException) {
+            throw new RuntimeException("Failed to initialize descriptors, this should never happen. ", anException);
+        }
+
+    }
+    //</editor-fold>
+
+    //<editor-fold desc="Public static methods">
+    /**
+     * Returns all available descriptors.
+     *
+     * @return All available descriptors
+     */
+    public static Descriptor[] getAllDescriptors(){
+        return values();
+    }
+
+    /**
+     * Returns all available fingerprint descriptors.
+     *
+     * @return All available fingerprint descriptors
+     */
+    public static Descriptor[] getAllFingerprints() {
+        return java.util.Arrays.stream(values())
+                .filter(Descriptor::isFingerprint)
+                .toArray(Descriptor[]::new);
+    }
+
+    /**
+     * Returns specified available descriptors.
+     * Note: If both speed flags are false, an empty array is returned. If both speed flags are true, then all descriptors are returned.
```

Comment:
Update the doc here. E.g. the third flag makes it more complex than it is stated here.